crfs.go

```diff
@@ -1060,6 +1060,10 @@ func (h *nodeHandle) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse
 		if err != nil {
 			return err
 		}
+		if nr == 0 {
+			// Truncate unncessary data
+			chunkData = chunkData[req.Offset:]
```

Comment:
This doesn't seem right in general.

This is in a loop (line 1050) is will keep filling, but req.Offset should only be used for the first chunk. And even then, only the part between ce.ChunkOffset and req.Offset should probably be used.

If they wanted overall req.Offset 100 but the 2nd chunk of the file is at offset 90, then what you really want here is offset 10.  You don't just want 100 each time.

This could probably use some tests, like the ones in the stargz package.

Reply:
Thank you for your review. Right. I fixed offset calculation and added some checks in 15a2736.